tests/integration/tests/test_network.py

```diff
@@ -78,20 +36,28 @@ def test_network(session_instance: harness.Instance):
         [
             "k8s",
             "kubectl",
-            "exec",
-            "-it",
-            cilium_pod["metadata"]["name"],
-            "-n",
-            "kube-system",
-            "-c",
-            "cilium-agent",
-            "--",
-            "cilium",
-            "endpoint",
-            "list",
+            "get",
+            "pod",
+            "-l",
+            "app=nginx",
             "-o",
             "json",
         ],
         capture_output=True,
     )
-    assert "nginx" in p.stdout.decode().strip()
+
+    out = json.loads(p.stdout.decode())
+
+    assert len(out["items"]) > 0, "No NGINX pod found"
+    podIP = out["items"][0]["status"]["podIP"]
```

Comment:
this needs some sort of retry in case the pod is not yet ready?

Reply:
Sure, lines 11-33 should wait until the pod reaches `condition=ready`. Do you want to retry the other parts of the test as well?

Reply:
The pod should always be ready at this point. As Mateo already said, we wait for the pod to come up further above.